packages/dd-trace/src/appsec/iast/taint-tracking/index.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const { storage } = require('../../../../../datadog-core')
+const iastContextFunctions = require('../iast-context')
+const log = require('../../../log')
+const { enableRewriter, disableRewriter } = require('./rewriter')
+const IAST_TRANSACTION_ID = Symbol('_dd.iast.transactionId')
+
+let TaintedUtils
+try {
+  TaintedUtils = require('@datadog/native-iast-taint-tracking')
+} catch (e) {
+  log.error(e)
+}
+
+const noop = function (res) { return res }
+const TaintTrackingDummy = {
+  plusOperator: noop
+}
+
+const TaintTracking = {
+  plusOperator: function (res, op1, op2) {
+    try {
+      if (typeof res !== 'string' ||
+        (typeof op1 !== 'string' && typeof op2 !== 'string')) { return res }
+
+      const store = storage.getStore()
+      const iastContext = iastContextFunctions.getIastContext(store)
+      const transactionId = iastContext && iastContext[IAST_TRANSACTION_ID]
+      if (transactionId) {
+        res = TaintedUtils.concat(transactionId, res, op1, op2)
+      }
+    } catch (e) {
+      log.debug(e)
+    }
+    return res
+  }
+}
+
+const createTransaction = function (id, iastContext) {
+  if (id && iastContext) {
+    const transactionId = TaintedUtils.createTransaction(id)
```

Comment:
All calls to native code should be try catched. We can't throw an unexpected exception if something fails in the native implementation.

Reply:
Should be try-catched here? or callers to these functions?

Reply:
I think that this is the point to `try catch` it, `createTransaction` and `removeTransaction` will be called only in one point, but it will be easy to forget to trycatch the methods like `newTaintedString`.

Reply:
We have decided that the module will return a Dummy objects, so try catch is not necessary, I'm going to resolve this conversation